youtube_dlc/downloader/external.py

```diff
@@ -189,22 +202,85 @@ class Aria2cFD(ExternalFD):
 
     def _make_cmd(self, tmpfilename, info_dict):
         cmd = [self.exe, '-c']
-        cmd += self._configuration_args([
-            '--min-split-size', '1M', '--max-connection-per-server', '4'])
-        dn = os.path.dirname(tmpfilename)
-        if dn:
-            cmd += ['--dir', dn]
-        cmd += ['--out', os.path.basename(tmpfilename)]
+        # cmd += self._configuration_args([
+        #    '--min-split-size', '1M', '--max-connection-per-server', '4'])
+        # dn = os.path.dirname(tmpfilename)
+        # if dn:
+        #     cmd += ['--dir', dn]
+        if 'url_list' not in info_dict:
+            cmd += ['--out', tmpfilename]
         if info_dict.get('http_headers') is not None:
             for key, val in info_dict['http_headers'].items():
                 cmd += ['--header', '%s: %s' % (key, val)]
         cmd += self._option('--interface', 'source_address')
         cmd += self._option('--all-proxy', 'proxy')
         cmd += self._bool_option('--check-certificate', 'nocheckcertificate', 'false', 'true', '=')
         cmd += self._bool_option('--remote-time', 'updatetime', 'true', 'false', '=')
-        cmd += ['--', info_dict['url']]
+        if 'url_list' in info_dict:
+            cmd += self._configuration_args(['--file-allocation', 'none', '--download-result', 'hide', '--uri-selector', 'inorder', '--console-log-level', 'warn', '-x16', '-j16', '-s16'])
+
+            url_list_file = '%s_urls.txt' % tmpfilename
+
+            url_list = []
+            for [i, url] in enumerate(info_dict['url_list']):
+                tmpsegmentname = '%s_%s.fragment' % (tmpfilename, i)
+                url_list.append('%s\n\tout=%s' % (url, tmpsegmentname))
+
+            with open(url_list_file, 'w') as f:
+                f.write('\n'.join(url_list))
```

Comment:
We need to handle what should happen if the file already exists or we get an `OSError`

Reply:
No other `with open` statement has an OSError handler, and 'w' mode overwrites the existing file.

Reply:
You are right. My bad